src/rpc/blockchain.cpp

```diff
@@ -139,17 +172,29 @@ UniValue blockToJSON(const CBlock& block, const CBlockIndex* tip, const CBlockIn
     result.pushKV("versionHex", strprintf("%08x", block.nVersion));
     result.pushKV("merkleroot", block.hashMerkleRoot.GetHex());
     UniValue txs(UniValue::VARR);
-    for(const auto& tx : block.vtx)
-    {
-        if(txDetails)
-        {
+
+    if (verbosity >= 2) {
+        const CBlockUndo blockUndo = GetUndoChecked(blockindex);
+        for (size_t i = 0; i < block.vtx.size(); ++i) {
+
+            const auto& tx = block.vtx.at(i);
+            const CTxUndo* ptr_txundo;
```

Comment:
Error:
```
error C4703: potentially uninitialized local pointer variable 'ptr_txundo'
```
Suggestion:
```cpp
const CTxUndo* tx_undo = tx->IsCoinBase() ? nullptr : &block_undo.vtxundo.at(i - 1);
```

Reply:
Could (should?) probably just use `i` instead of IsCoinBase